### PR TITLE
InstallerConfig

### PR DESCRIPTION
**Pull request by leebm3**

### CODE REVIEW

File: lib/installer.ts [hunk 1]
```diff
@@ -46,14 +46,14 @@ const step = async (title: string, f: InstallerStepFn) => {
...
+export const startInstaller = async (steps: Step[], config: InstallerConfiguration = {}) => {
+  if (config.header) {
+    console.info(config.header);
   }
-
+  const logger = new Logger(config.loggerFileName)
   for (const { title, f } of steps) {
     const res = await step(title, f);
-    logToFile(JSON.stringify({ ...res, title }, null));
+    logger.logToFile(JSON.stringify({...{"title" : title}, ...res}, null));
```
Comment (drorwolmer) [inline lib/installer.ts:56]:
```js
{title, ...res}
```

**Review by drorwolmer — CHANGES_REQUESTED** [review]

File: lib/logsHandler.ts [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+    constructor(name?: string) {
+        this.name = `${name || "installer"}-${moment(new Date()).format('YYYY-MM-DDThmmss')}.log`
+    }
+
+    logPath() {
+        return join(tmpdir(), this.name)
+    }
+
+    logToFile(message: string) {
+        appendFileSync(this.logPath(), `[INFO] ${new Date().toISOString()} ${message}\n`)
```
Comment (drorwolmer) [inline lib/logsHandler.ts:18]: no need to call `logpath()` every time `logToFile()` is called, it should be calculated on the constructor

**Comment by leebm3** [comment]
@drorwolmer 
Added the proxy Validation step if proxy wasn't set correctly in env.
Added proxy get from env

File: lib/network.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import clc from "cli-color";
 import httpsProxyAgent, { HttpsProxyAgent } from "https-proxy-agent";
 import { URL } from "url";
 import * as net from "net";
+import { env } from 'process';
```
Comment (drorwolmer) [inline lib/network.ts:9]: unneeded

**Review by drorwolmer — CHANGES_REQUESTED** [review]
also make sure you use `prettier` on all this stuff

File: lib/logsHandler.ts [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
-  "YYYY-MM-DDThmmss"
-)}.log`;
-export const LOG_PATH = join(tmpdir(), LOG_FILE_NAME);
+export class Logger {
+    name: string;
+    pathToLogFile: string;
+    
+    constructor(name?: string) {
+        this.name = `${name || "installer"}-${moment(new Date()).format('YYYY-MM-DDThmmss')}.log`
+        this.pathToLogFile = join(tmpdir(), this.name)
```
Comment (drorwolmer) [inline lib/logsHandler.ts:12]: `this.name` is not really needed, just the `filepath` attribute is used (internally)
refactor this

**Review by drorwolmer — APPROVED** [review]

**Pull request merged**